tests/scripts/all.sh

```diff
@@ -1262,6 +1262,25 @@ component_test_cmake_out_of_source () {
     unset MBEDTLS_ROOT_DIR
 }
 
+component_test_cmake_as_subdirectory () {
+    msg "build: cmake 'as-subdirectory' build"
+    MBEDTLS_ROOT_DIR="$PWD"
+
+    # Remove build artifacts if any from previous builds.
+    rm -rf programs/test/cmake_subproject/build
```

Comment:
Suggestion: Perhaps is worth adding this to the `cleanup()` instead of here?

Reply:
This step is generally not needed. Only this tests makes a mess there, so only this one cleans up its mess. That was my thinking.

Reply:
I do not feel very strongly about it, so we can leave it like that.

My suggestion was to ensure that this cleanup is executed regardless if there is an early failure in the test. For example, if we were to leave it here and a subsequent command before line 1238 fails, then the script might terminate and not do the cleanup step in line 1238. In contrast, I would expect the cleanup() call to be done upon exit

Reply:
OK, moved into `cleanup()`.